integration-tests/jpa-postgresql-withxml/src/main/java/io/quarkus/it/jpa/postgresql/JPAFunctionalityTestEndpoint.java

```diff
@@ -0,0 +1,236 @@
+            checkWrittenXmlObject(con);
+        }
+    }
+
+    private void checkWrittenXmlObject(Connection con) throws SQLException {
+        try (Statement stmt = con.createStatement()) {
+            final ResultSet resultSet = stmt.executeQuery("SELECT val FROM xmltest");
+            final boolean next = resultSet.next();
+            if (!next) {
+                throw new IllegalStateException("Stored XML element not found!");
+            }
+            final String storedValue = resultSet.getString(1);
+            if (storedValue == null) {
+                throw new IllegalStateException("Stored XML element was loaded as null!");
+            }
+            String expectedMatch = "<?xml version=\"1.0\" standalone=\"no\"?><root><ele>1</ele><ele>2</ele></root>";
+            if (!expectedMatch.equals(storedValue)) {
+                throw new IllegalStateException("Stored XML element is not matching expected value of '" + expectedMatch
+                        + "', but was '" + storedValue + "'");
+            }
+        }
+    }
+
+    private void writeXmlObject(Connection con) throws SQLException, TransformerException {
+        TransformerFactory factory = TransformerFactory.newInstance();
+        final String _xmlDocument = "<root><ele>1</ele><ele>2</ele></root>";
+        Transformer identityTransformer = factory.newTransformer();
+        try (PreparedStatement ps = con.prepareStatement("INSERT INTO xmltest VALUES (?,?)")) {
+            SQLXML xml = con.createSQLXML();
+            Result result = xml.setResult(DOMResult.class);
+
+            Source source = new StreamSource(new StringReader(_xmlDocument));
+            identityTransformer.transform(source, result);
+
+            ps.setInt(1, 1);
+            ps.setSQLXML(2, xml);
+            ps.executeUpdate();
+        }
+    }
+
+    private void createXmlSchema(Connection con) throws SQLException {
+        Statement stmt = con.createStatement();
+        stmt.execute("CREATE TEMP TABLE xmltest(id int primary key, val xml)");
+        stmt.close();
+    }
+
+    private void deleteXmlSchema(Connection con) {
+        try {
+            Statement stmt = con.createStatement();
+            stmt.execute("DROP TABLE xmltest");
+            stmt.close();
+        } catch (SQLException throwables) {
+            //ignore this one
+        }
+    }
+
+    /**
+     * Lists the various operations we want to test for:
+     */
+    private static void doStuffWithHibernate(EntityManagerFactory entityManagerFactory) {
+
+        //Store some well known Person instances we can then test on:
+        storeTestPersons(entityManagerFactory);
+
+        //Load all persons and run some checks on the query results:
+        verifyListOfExistingPersons(entityManagerFactory);
+
+        //Try a JPA named query:
+        verifyJPANamedQuery(entityManagerFactory);
+    }
+
+    private static void useXMLParsing(EntityManagerFactory entityManagerFactory) {
```

Comment:
I believe this method is no longer used.

Reply:
[removed it]